src/ddmd/codegenASTFamily.d

```diff
@@ -0,0 +1,42 @@
+module ddmd.codegenASTFamily;
+
+import ddmd.arraytypes;
+import ddmd.dclass;
+import ddmd.dimport;
+import ddmd.dstruct;
+import ddmd.globals;
+import ddmd.identifier;
+import ddmd.mtype;
+
+struct CodegenASTFamily
+{
+    StructDeclaration iStructDeclaration(Loc loc, Identifier id)
+    {
+        return new StructDeclaration(loc, id);
+    }
+
+    UnionDeclaration iUnionDeclaration(Loc loc, Identifier id)
+    {
+        return new UnionDeclaration(loc, id);
+    }
+
+    BaseClass* iBaseClass(Type t)
+    {
+        return new BaseClass(t);
+    }
+
+    InterfaceDeclaration iInterfaceDeclaration(Loc loc, Identifier id, BaseClasses* baseclasses)
+    {
+        return new InterfaceDeclaration(loc, id, baseclasses);
+    }
+
+    ClassDeclaration iClassDeclaration(Loc loc, Identifier id, BaseClasses* baseclasses, Dsymbols* members, bool inObject)
+    {
+        return new ClassDeclaration(loc, id, baseclasses, members, inObject);
+    }
+
+    Import iImport(Loc loc, Identifiers* a, Identifier id, Identifier aliasId, int isstatic)
+    {
+        return new Import(loc, a, id, aliasId, isstatic);
+    }
+}
```

Comment:
Great, but no need to tie types with the allocation method. Just this should suffice:

```d
alias StructDeclaration = ddmd.dstruct.StructDeclaration;
alias UnionDeclaration = ddmd.dstruct.UnionDeclaration;
alias BaseClass = ddmd.dclass.BaseClass;
...
```

Then the rest of the code could continue using `new ASTFamily.StructDeclaration;` etc.

If there's a need, we may factor out the allocation and construction mechanisms as well in a later pass.